aten/src/ATen/native/vulkan/api/Context.cpp

```diff
@@ -0,0 +1,281 @@
+        "VK_LAYER_GOOGLE_unique_objects",
+        "VK_LAYER_GOOGLE_threading",
+        "VK_LAYER_LUNARG_object_tracker",
+        "VK_LAYER_LUNARG_core_validation",
+        "VK_LAYER_LUNARG_parameter_validation",
+        "VK_LAYER_KHRONOS_validation",
+    };
+
+    for (const auto& requested_instance_layer : requested_instance_layers) {
+      for (const auto& layer : instance_layer_properties) {
+        if (strcmp(requested_instance_layer, layer.layerName) == 0) {
+          enabled_instance_layers.push_back(requested_instance_layer);
+          break;
+        }
+      }
+    }
+
+    uint32_t instance_extension_count = 0;
+    VK_CHECK(vkEnumerateInstanceExtensionProperties(
+        nullptr, &instance_extension_count, nullptr));
+
+    std::vector<VkExtensionProperties> instance_extension_properties(
+        instance_extension_count);
+
+    VK_CHECK(vkEnumerateInstanceExtensionProperties(
+        nullptr, &instance_extension_count, instance_extension_properties.data()));
+
+    constexpr std::array<const char*, 1> requested_instance_extensions{
+      VK_EXT_DEBUG_REPORT_EXTENSION_NAME,
+    };
+
+    for (const auto& requested_instance_extension : requested_instance_extensions) {
+      for (const auto& extension : instance_extension_properties) {
+        if (strcmp(requested_instance_extension, extension.extensionName) == 0) {
+          enabled_instance_extensions.push_back(requested_instance_extension);
+          break;
+        }
+      }
+    }
+  }
+
+  constexpr VkApplicationInfo application_info{
+    VK_STRUCTURE_TYPE_APPLICATION_INFO,
+    nullptr,
+    "PyTorch",
+    0,
+    "PyTorch",
+    0,
+    VK_API_VERSION_1_0,
+  };
+
+  const VkInstanceCreateInfo instance_create_info{
+    VK_STRUCTURE_TYPE_INSTANCE_CREATE_INFO,
+    nullptr,
+    0u,
+    &application_info,
+    enabled_instance_layers.size(),
+    enabled_instance_layers.data(),
+    enabled_instance_extensions.size(),
+    enabled_instance_extensions.data(),
+  };
+
+  VkInstance instance{};
+  VK_CHECK(vkCreateInstance(&instance_create_info, nullptr, &instance));
+
+  if (enable_validation_layers) {
+    const VkDebugReportCallbackCreateInfoEXT debugReportCallbackCreateInfo{
+      VK_STRUCTURE_TYPE_DEBUG_REPORT_CALLBACK_CREATE_INFO_EXT,
+      nullptr,
+      VK_DEBUG_REPORT_INFORMATION_BIT_EXT |
+        VK_DEBUG_REPORT_WARNING_BIT_EXT |
+        VK_DEBUG_REPORT_PERFORMANCE_WARNING_BIT_EXT |
+        VK_DEBUG_REPORT_ERROR_BIT_EXT |
+        VK_DEBUG_REPORT_DEBUG_BIT_EXT,
+      debug_report_callback_fn,
+      nullptr,
+    };
+
+    const auto vkCreateDebugReportCallbackEXT =
+        (PFN_vkCreateDebugReportCallbackEXT)vkGetInstanceProcAddr(
+            instance, "vkCreateDebugReportCallbackEXT");
+
+    TORCH_CHECK(
+        vkCreateDebugReportCallbackEXT,
+        "Could not load vkCreateDebugReportCallbackEXT");
+
+    VkDebugReportCallbackEXT debug_report_callback{};
+    VK_CHECK(vkCreateDebugReportCallbackEXT(
```

Comment:
debugReportCallback needs explicit destructor like:
```
    const auto func = (PFN_vkDestroyDebugReportCallbackEXT)vkGetInstanceProcAddr(
        instance_, "vkDestroyDebugReportCallbackEXT");
    if (func) {
      func(instance_, debugReportCallback, nullptr);
    }
```
( https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkDestroyDebugReportCallbackEXT.html )

Here it is only as temporary variable in `create_instance()`
Does it need some special deleter function?

Reply:
Thanks for the catch.

Reply:
By the way, VK_EXT_validation_flags is deprecated.  At one point we should replace it with VK_EXT_debug_utils.  I'll do that later on.